etcdctl/command/util.go

```diff
@@ -213,9 +213,13 @@ func mustNewClient(c *cli.Context) client.Client {
 
 			// fail-back to try sync cluster with peer API. this is for making etcdctl work with etcd 0.4.x.
 			// TODO: remove this when we deprecate the support for etcd 0.4.
-			eps, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
-			if serr != nil {
-				handleError(ExitServerError, serr)
+			eps, trerr, serr := syncWithPeerAPI(c, ctx, hc.Endpoints())
```

Comment:
this is the backward compatibility path. If the original error is not client.ErrNoEndpoints, we probably should also exit with badconnection.